aeternity/aeternity
# Minerva accounts.json

PT [Integrate token file for mainnet hard fork](https://www.pivotaltracker.com/story/show/164104629)
This is the list of accounts migrated during `phase1` of the token migration.

## How to test
This is a publicly available information and everybody must be able to verify it in a trustless manner. Aeternity provides two different scripts that could be run in order to produce the file.

### Prerequisites
My setup is
#### NPM
```
$ npm -v
6.8.0
```
#### Node.js
```
$ node -v
v11.10.0
```
I've tested both scripts to yield the same result with this setup. Using older versions might result in the script not running at all.

### Using a Infura WebSocket provider
Code: [aeternity/initial-ae-balances-generator](https://github.com/aeternity/initial-ae-balances-generator)
It requires one's own WebSocket provider

Example log
```
$ node initial-balances-generator.js -p <provider> -d 1
Starting..
574 burn events to be found
================================================================================================================================================================================- 571/574 Burn events found.
File saved: accounts.json.
```

### Using Etherscan.io
Code: aeternity/initial-ae-balances-generator-etherscan#6
It requires a valid Etherscan.io API key which can be requested on http://etherscan.io/

Example log
```
$ node initial-balances-generator-etherscan.js -k XXXXXXXXXXXXXXXXXXXXXXXX
Getting Logs for phase 1 from block 6772802 to 7225342 
Check the following urls for 'from' and 'to' block: 
 end of phase 0 
	 https://etherscan.io/tx/0x6dd35f57ac61436c9ee7ebc2cd0be830506965aca072eb23dc0e4e4b6456cbfb 

 end of token migration phase 1: 
	https://etherscan.io/tx/0xed2415a9e8026e042c58e701bfbe218f858ad71bfa9796c9b8936b8faae83155

Getting Logs from block 6772802 to 6885937. Please wait...
Getting Logs from block 6885937 to 6999072. Please wait...
Getting Logs from block 6999072 to 7112207. Please wait...
Getting Logs from block 7112207 to 7225342. Please wait...
Processed 574 events with 325 individual aeternity accounts
Collected 27051422.546127651538826703 AE Tokens in this phase

./accounts.json written to disk
verify sha256 hash of accounts json with this command:
	$ shasum -a 256 accounts.json
```
## Checking the file
Both scripts produce the same `accounts.json` file
```
$ shasum -a 256 accounts.json
c63483d54dc4de031265a239fa8aafe1d9d5a36728410df25bec52e2f15993fb  accounts.json
```
